### PR TITLE
Wip/6.0 re enabling tests 

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java [hunk 1]
```diff
@@ -47,10 +48,10 @@ public SelfRenderingFunctionSqlAstExpression convertToSqlAst(SqmToSqlAstConverte
 		return new SelfRenderingAggregateFunctionSqlAstExpression(
 				getFunctionName(),
 				getRenderingSupport(),
-				resolveSqlAstArguments( getArguments(), walker ),
+				resolveSqlAstArguments( getArguments(), (BaseSqmToSqlAstConverter) walker ),
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java:51]: Seems unnecessary, or am I missing something?

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java [hunk 2]
```diff
@@ -47,10 +48,10 @@ public SelfRenderingFunctionSqlAstExpression convertToSqlAst(SqmToSqlAstConverte
...
 				filter == null ? null : (Predicate) filter.accept( walker ),
 				resultType,
-				getMappingModelExpressable( walker, resultType )
+				getMappingModelExpressable( (BaseSqmToSqlAstConverter) walker, resultType )
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java:54]: Seems unnecessary, or am I missing something?

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java [hunk 3]
```diff
@@ -60,7 +62,21 @@ public FunctionRenderingSupport getRenderingSupport() {
 
 		final ArrayList<SqlAstNode> sqlAstArguments = new ArrayList<>( sqmArguments.size() );
 		for ( SqmTypedNode<?> sqmArgument : sqmArguments ) {
-			sqlAstArguments.add( (SqlAstNode) ( (SqmVisitableNode) sqmArgument ).accept( walker ) );
+			if ( sqmArgument instanceof SqmParameter ) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java:65]: What's the reason for this? There are legitimate uses of functions where a tuple should be passed in rather than the expanded list.

**Review by beikov — CHANGES_REQUESTED** [review]
Please also rebase against wip/6.0 to fix the issues we had with running on Java 8.

File: hibernate-core/src/test/java/org/hibernate/orm/test/connections/LazyLoadingConnectionCloseTest.java [hunk 4]
```diff
@@ -97,39 +98,44 @@ public void setUp() {
 		} );
 	}
 
+	@AfterAll
+	public void tearDown() {
+		inTransaction( entityManager -> {
+			entityManager.createNativeQuery( "delete from ChildEntity" ).executeUpdate();
+			entityManager.createNativeQuery( "delete from SimpleEntity" ).executeUpdate();
```
[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/orm/test/connections/LazyLoadingConnectionCloseTest.java:105]: yeap you are right. going to fix it. Thanks

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java:51]: you are right, going to remove the cast. Thanks

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java (hunk 2 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmAggregateFunction.java:54]: some here, thanks

File: hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java (hunk 3 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java:65]: I changed this to fix `org.hibernate.orm.test.ordered.HqlOrderByIdsTest` where the following query
```
List<Person> persons = entityManager.createQuery(
					"SELECT p " +
							"FROM Person p " +
							"WHERE p.id IN (:ids) " +
							"ORDER BY FIELD(id, :ids) ", Person.class )
					.setParameter( "ids", Arrays.asList( 3L, 1L, 2L ) )
					.getResultList();
```
was failing, when we try to create the `JdbcParameterBindings` in the `SqmUtil#createJdbcParameterBindings` we have an assertion failure (line 233) `assert expansions.size() == bindValues.size() - 1` because for the second one of the `:ids`   `bindValues.size()` is 0

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java:65]: by the way not sure that my solution is the correct /better one

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java:65]: Hmm, IMO the function should in the rendering expand the tuple manually instead of doing this expansion in the general rendering code.

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java [hunk 5]
```diff
@@ -785,7 +785,7 @@ public Expression resolveSqlExpression(
 				if ( sqmAssignment.getValue() instanceof SqmParameter ) {
 					final SqmParameter<?> sqmParameter = (SqmParameter<?>) sqmAssignment.getValue();
 
-					consumeSqmParameter(
+					consumeSingleSqmParameter(
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:788]: Was this change on purpose?

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:788]: we renamed the method

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:788]: I know, but it seems like we might have forgotten to replace this one with `consumeSqmParameter` again. Nevermind though, it seems odd to have a multi-valued parameter in the SET clause.

**Review by beikov — APPROVED** [review]
